packages/runtime/container-runtime/src/id-compressor/README.md

```diff
@@ -1,3 +1,69 @@
-# ID Compressor
+# **Distributed ID Allocator**
 
-An [IdCompressor](./idCompressor.ts) generates unique identifiers that may be efficiently compressed/decompressed and synchronized with other IdCompressors.
+A library which generates small number representations of arbitrary non-colliding Version 4 UUIDs (stable IDs) across multiple sessions in a network. This scheme enables a distributed application to utilize the global uniqueness guarantees of UUIDs while maintaining the performance advantages of small integers.
+
+## **Overview**
+
+The distributed ID allocation scheme allows clients to use, store, and reference small numbers that map to UUIDs. The primary benefits are improved memory efficiency and reduced storage size, as well as improved runtime performance when used as keys in collections or other data structures in place of UUIDs.
+
+The scheme requires a total order for ID allocation, so the allocator is designed to work within a service that provides centralized total order broadcast.
+
+## **Sessions**
+
+A session can be thought of as a unique identifier for a single allocator. Even if an allocator is taken offline, serialized to disk, and rehydrated back, it's considered the same session if it has the same session ID.
+
+## **Document**
+
+The set of IDs generated by all Sessions contributing to the same distributed collection (e.g. an individual collaborative document). Uniqueness of allocated IDs is limited to the document context (unless in their UUID form), and thus IDs in their compressed form are not portable between documents.
+
+## **Generation API**
+
+New IDs are generated via synchronous API on the allocator. The API returns a small number representation of the UUID with the following guarantees:
+
+- The ID can be synchronously converted into a full UUID at any time. This is required if global uniqueness (across documents) is needed.
+- Prior to the service being notified of the ID's creation, it is unique within the scope of the session (e.g. it can collide with IDs allocated by a different allocator in the same network).
```

Comment:
```suggestion
- Prior to the service being notified of the ID's creation, it is unique within the scope of the session (i.e. the new ID will not collide with other IDs from the same allocator, but it can collide with IDs from a different allocator in the same network).
```
Statement in parentheses confused me because it's clarifying the preceding point, but doing it by counterexample without indicating so